### PR TITLE
configure script for ENFNORMALIZ

### PR DESCRIPTION
**Pull request by videlec**
solve #42.

- [x] make distributions (on PyPI): there is a target `sdist` in the `Makefile`. Some of the configure script must be shipped in the tarball and this has been added to `MANIFEST.in`.
- [x] The PyNormaliz distribution does not depend on autoconf. Since `configure.ac` is not shipped in the source tarballs (see `MANIFEST.in`) the target `make configure` is void (in the distribution).
- [x] The `NORMALIZ_LOCAL_DIR` is not supported anymore... To be fixed later.
- [x] There were weird `Makefile` targets. With the current version, to install with python2 do `PYTHON=python2 make install` and to install with python3 do `PYTHON=python3 make install`

### CODE REVIEW

**Comment by videlec** [comment]
Perhaps, you want to check for Normaliz version in the configure script. Since you require it in the README that would be a natural thing to do.

File: setup.py [hunk 1]
```diff
@@ -56,6 +42,31 @@ def run(self):
         finally:
             os.chdir(old_path)
 
+class build(_build):
+    def run(self):
+        """
+        Run ./configure first and update libraries depending
+        on the generated output.
+        """
+        subprocess.check_call(["make", "configure"])
```
Comment (sebasguts) [inline setup.py]: I am a bit confused by this. `make configure` just generates the `configure` script, but does not run it. Is that intentional? Do you mean do execute `make dist` here, which executes `configure`?

File: setup.py [hunk 2]
```diff
@@ -67,8 +78,9 @@ def run(self):
     ext_modules = [ Extension( "PyNormaliz_cpp",
                               [ "NormalizModule.cpp" ],
                               extra_compile_args=['-std=c++11'],
+                              libraries=libraries,
                               **extra_kwds) ],
     
     package_data = {'': [ "COPYING", "GPLv2", "Readme.md" ] },
-    cmdclass = {'test': TestCommand},
+    cmdclass = {'build': build, 'test': TestCommand},
```
Comment (sebasguts) [inline setup.py]: Do you want to do `build` here` C dependencies are usually build via `build_ext`, so maybe we should overload that command instead? I am not sure though,

File: setup.py (hunk 1 above)
Comment (videlec) [reply setup.py]: It is intentional (and standard). `make X` builds `X`. It does not run it.

File: setup.py (hunk 2 above)
Comment (videlec) [reply setup.py]: It is also a possibility but apparently `pip` does not call `build_ext`. I am trying to understand this...

**Comment by videlec** [comment]
Do I assume that you agree on the principle of having a configure script generated by autoconf?

File: setup.py (hunk 1 above)
Comment (videlec) [reply setup.py]: I can add a `RUN_configure` target. But it is better to separate the steps in the `setup.py` (more meaningful error messages).

File: setup.py (hunk 2 above)
Comment (videlec) [reply setup.py]: Sorry, pip does call `build_ext`.

Comment (videlec) [reply setup.py]: What is confusing is that `pip` creates a temporary directory, copy all files, and then run the various setup steps.

Comment (videlec) [reply setup.py]: Changed in 6a60f7d

**Comment by videlec** [comment]
The test failures on Travis are tracked #53. The configure script is not to blame.

File: setup.py (hunk 1 above)
Comment (sebasguts) [reply setup.py]: Yep, I understand. What I do not understand is the logic here.

You execute `make configure` which builds the configure script, but does not execute it. I thought you wanted to execute the configure script, which is not done in this `setup.py` step. So the `configure` never gets executed by `setup.py` (or at least it seems like it)

**Comment by sebasguts** [comment]
> Do I assume that you agree on the principle of having a configure script generated by autoconf?

Yes, overall I am fine with this. It is another dependency of course, and maybe pure python ways would be preferable, but I think it is okay.

File: setup.py (hunk 1 above)
Comment (videlec) [reply setup.py]: Yes, it is the lines just after
```
        try:
            subprocess.check_call(["sh", "configure"])
        except subprocess.CalledProcessError:
            subprocess.check_call(["cat", "config.log"])
            raise
```

**Comment by videlec** [comment]
Can I have your opinion on the two unchecked items in the pull request description?

**Comment by videlec** [comment]
Tests pass (including installation from generated source tarball). Remains just the two unchecked boxes...

**Comment by sebasguts** [comment]
@videlec I will take care of the makefile flags tomorrow. They can most likely be dropped.

I am not a huge fan to drop support of the env variable completely.
The flags that setuptools understand are insufficient (because the rpaths are not set correctly), and setting all env variables for the compiler is quite a hassle. I will readd support for the variable.

File: setup.py (hunk 1 above)
Comment (sebasguts) [reply setup.py]: Ah, okay, missed that, sorry :)

**Comment by videlec** [comment]
Then I am done here :-)

**Comment by sebasguts** [comment]
Thank you very much

**Pull request merged**